sympy/matrices/common.py

```diff
@@ -753,11 +884,39 @@ def size(m):
                         diag_entries[(i + row_pos, j + col_pos)] = m[i, j]
                 row_pos += m.rows
                 col_pos += m.cols
-            else:
-                # in this case, we're a single value
+            elif not isinstance(m, dict):
+                # in this case we're a single value
                 diag_entries[(row_pos, col_pos)] = m
                 row_pos += 1
                 col_pos += 1
+            else:
+                # in this case we're a dict
+                standardize_values(m)
+                rmax, cmax = m.pop('size')
+                for key, value in m.items():
+                    r_p = 0 if key >= 0 else -key
+                    c_p = 0 if r_p else key
+                    while (r_p < rmax and c_p < cmax):
+                        D = r_p + row_pos, c_p + col_pos
+                        if isinstance(value, list):
+                            for i in range(len(value)):
+                                diag_entries[D] = value[i]
+                                D = D[0] + 1, D[1] + 1
+                            r_p = rmax
```

Comment:
these two lines can just be `break` to get out of while loop

Reply:
Yes, that can be done, I edited it to `break`.